packages/astro/src/core/routing/manifest/create.ts

```diff
@@ -706,6 +719,20 @@ export function createRouteManifest(
 	};
 }
 
+export function resolveInjectedRoute(entrypoint: string, root: URL, cwd?: string) {
+	let resolved;
+	try {
+		resolved = require.resolve(entrypoint, { paths: [cwd || fileURLToPath(root)] });
```

Comment:
But besides my shenanigans. I think this won't work for routes injected by transitive dependencies when using a strict package manager like so:

```
Project - has -> Integration A - adds -> Integration B - injects -> the route
```
where the package defining `Integration B` is not a direct dependency of the project.

This breaks currently, so it's fine as is. Just pointing it out that it will break here now instead of inside of Vite so the error will change.

Reply:
This code is not new, it just moved